src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -416,7 +416,29 @@ private static bool TryGetRepresentativeTypeNameFromValue(object value, out PSTy
                 value = PSObject.Base(value);
                 if (value != null)
                 {
-                    type = new PSTypeName(value.GetType());
+                    var typeObject = value.GetType();
+                    
+                    if (typeObject.FullName.Equals("System.Management.Automation.PSObject", StringComparison.Ordinal))
+                    {
+                        var psobjectPropertyList = new List<PSMemberNameAndType>();
+                        foreach (var property in ((PSObject)value).Properties)
```

Comment:
Why do we get _only_ properties?

Reply:
What else are you interested in? Methods? I realize the type is called "PSMemberNameAndType" but it doesn't seem like the synthetic type actually handles methods. I can add methods if you want though.

Reply:
I guess users will wonder if they add methods with Add-Member and we don't list them.

Reply:
I just tested it locally. If I add methods they show up when doing `$Var | Select <Tab>` which obviously shouldn't happen. The issue is that there's no way to differentiate methods and properties in the synthetic type. You are of course right that it would make sense to show them but I don't think it's needed for this PR.